presto-main/src/main/java/com/facebook/presto/metadata/AllNodes.java

```diff
@@ -27,14 +27,22 @@
     private final Set<InternalNode> shuttingDownNodes;
     private final Set<InternalNode> activeCoordinators;
     private final Set<InternalNode> activeResourceManagers;
+    private final Set<InternalNode> activeCatalogServers;
```

Comment:
Are we sure we wanna make catalog server a separated tier from the beginning. Is it possible to make the module part of the resource manager as a first step? Having too many tiers will make our topology quite bit complicated.